clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -854,6 +854,14 @@ boolean reconcile() {
         return true;
     }
 
+    long getExpirationTimeForTimeout(final long timeoutMs) {
+        long expiration = System.currentTimeMillis() + timeoutMs;
+        if (expiration < 0) {
+            return Long.MAX_VALUE;
```

Comment:
Not sure if we'll ever enter this branch?

Reply:
It would be the case if a Long.MAX_VALUE is passed as timeout, give that it would overflow. It's not expected to be the case with the rebalance timeout, but just to keep the calculation right, no matter the timeout used.